api/account/api.go

```diff
@@ -111,26 +112,30 @@ func ConvertToAPIAccount(ctx context.Context, userKc kc.UserRepresentation, logg
 		userRep.Locale = value
 	}
 	if values := userKc.GetAttribute(constants.AttrbAccreditations); len(values) > 0 {
-		var accreds []AccreditationRepresentation
-		for _, accredJSON := range values {
-			var accred AccreditationRepresentation
-			if json.Unmarshal([]byte(accredJSON), &accred) == nil {
-				// If accreditation is revoked, considers it as expired so that this version is backward compatible with backoffice
-				if accred.Revoked != nil && *accred.Revoked {
-					accred.Expired = accred.Revoked
-				} else {
-					accred.Expired = keycloakb.IsDateInThePast(accred.ExpiryDate)
-				}
-				accred.Revoked = nil
-				accreds = append(accreds, accred)
+		userRep.Accreditations = convertToAccreditations(ctx, values, logger)
+	}
+
+	return userRep
+}
+
+func convertToAccreditations(ctx context.Context, values []string, logger keycloakb.Logger) *[]AccreditationRepresentation {
+	var accreds []AccreditationRepresentation
+	for _, accredJSON := range values {
+		var accred AccreditationRepresentation
+		if json.Unmarshal([]byte(accredJSON), &accred) == nil {
+			// If accreditation is revoked, considers it as expired so that this version is backward compatible with backoffice
+			if accred.Revoked != nil && *accred.Revoked {
+				accred.Expired = accred.Revoked
 			} else {
-				logger.Warn(ctx, "msg", "Can't unmarshall JSON", "json", accredJSON)
+				accred.Expired = keycloakb.IsDateInThePast(accred.ExpiryDate)
```

Comment:
I imagine this a bug for a different topic, it might be good to have it in a specific PR to be able to track it in future ?

Reply:
No, it's just the diff which is not correctly computed by GitHub... the warning is still present (see L#135)